File: web/app/models/tile.js
import DS from 'ember-data';

var Tile = DS.Model.extend({
    'tile-template': DS.attr('string', {defaultValue: 'tt-01'}),
    'content-template': DS.attr('string', {defaultValue: 'ct-fill'}),
    animation: DS.attr('string', {defaultValue: 'ease-01'}),
    title: DS.attr('string'),
    style: DS.attr('string'),
    href: DS.attr('string'),
    size: DS.attr('string'),
    'tile-data' : DS.hasMany('tile-data', {embedded: 'always'})
});

export default Tile;

File: server/homepage.Api/Tile.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace homepage.Api
{
    public class Tile
    {
        [JsonProperty("tile-template")]
        public string Template { get; set; }

        [JsonProperty("content-template")]
        public string ContentTemplate { get; set; }

        [JsonProperty("animation")]
        public string Animation { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("style")]
        public string CssClass { get; set; }

        [JsonProperty("href")]
        public Uri Link { get; set; }

        [JsonProperty("size")]
        public string Size { get; set; }

        [JsonProperty("data")]
        public IEnumerable<TileContent> Content { get; set; }

        public Tile()
        {
            Template = "tt-01";
            ContentTemplate = "ct-fill";
            Animation = "ease-01";
            Size = "large";
        }
    }
}